dotnet/src/SemanticKernel.Abstractions/AI/ChatClient/KernelFunctionInvokingChatClient.cs

```diff
@@ -626,13 +106,15 @@ select ProcessFunctionCallAsync(
                 // Invoke each function serially.
                 for (int i = 0; i < functionCallContents.Count; i++)
                 {
-                    var result = await ProcessFunctionCallAsync(
+                    var functionResult = await this.ProcessFunctionCallAsync(
                         messages, options, functionCallContents,
                         iteration, i, captureCurrentIterationExceptions, isStreaming, cancellationToken).ConfigureAwait(false);
 
-                    results.Add(result);
+                    results.Add(functionResult);
 
-                    if (result.ShouldTerminate)
+                    // Differently from the original FunctionInvokingChatClient, as soon the termination is requested,
+                    // we stop and don't continue, if that can be parametrized, this override won't be needed
+                    if (functionResult.ShouldTerminate)
```

Comment:
If we were to parameterize this, I presume we'd add a public property to `FunctionInvokingChatClient` to do so, right?

I see benefits of adding that property alongside the other "policy" properties on the class that allow control over iterations and other aspects of how much work should be allowed to be done.

If we can come up with a good name for the property, I'd be in favor of adding it for MEAI GA.

Reply:
> If we can come up with a good name for the property, I'd be in favor of adding it for MEAI GA.

All ears to what name we can have for it or a `Policy` approach. What's your thoughts @SteveSandersonMS?